eval/builtin_fn_io.go

```diff
@@ -122,13 +125,48 @@ func toLines(fm *Frame, inputs Inputs) {
 	})
 }
 
+// Exported so that other packages can refer to it for proper nil
```

Comment:
* This doesn't need to be exported; it is not used in another other package.
* This function is only needed when "numberify" is enabled. Just name it `numberify` and avoid calling it when the option is not given.
* Instead of only handling marshaling of `Nil` here, implement [`Json.Marshaler`](https://godoc.org/encoding/json#Marshaler) for the `Nil` type so that it can be marshaled into JSON using the standard JSON encoding functions.

Reply:
I call it from Go functions that are implemented in a namespace I inject using the ExtraNS stuff -- that namespace is responsible for implementing in-shell access to a JSON baseed API that does not deal well with numbers and nulls magically transmuting to strings.  If this is not a good place for it, is there a better location?

Reply:
Elvish does not have a stable Go API. Helper functions like these are especially subject to change. If you want to use Elvish from another project, it is a better idea to duplicate those code in the consumer package. The other APIs you use are also subject to change, but less so.